corehq/apps/cloudcare/static/cloudcare/js/formplayer/menus/views.js

```diff
@@ -243,6 +245,43 @@ hqDefine("cloudcare/js/formplayer/menus/views", function () {
             };
         },
 
+        iconClick: function (e) {
+            e.stopImmediatePropagation();
+            const origin = window.location.origin;
+            const user = FormplayerFrontend.getChannel().request('currentUser');
+            const appId = formplayerUtils.currentUrlToObject().appId;
+            const currentApp = FormplayerFrontend.getChannel().request("appselect:getApp", appId);
+            // Confirms we are getting the app id, not build id
+            const currentAppId = currentApp.attributes["copy_of"] ? currentApp.attributes["copy_of"] : currentApp.attributes["_id"]
+            const domain = user.domain;
+            const caseId = this.model.get('id');
+            const fieldIndex = $(e.currentTarget).parent().index();
+            const urlTemplate = this.options.endpointActions[fieldIndex]['urlTemplate'];
+            const actionUrl = origin + urlTemplate
+                .replace("{domain}", domain)
+                .replace("{appid}", currentAppId)
+                .replace("{case_id}", caseId);
+            e.target.className += " disabled";
+            document.location = actionUrl;
```

Comment:
how come we are redirecting now?

Reply:
this was an intermediate step. The background ajax request wasn't updating cases and was giving a false positive response. Working on the background request now, and I'll be removing the redirect next.

Reply:
fixed here https://github.com/dimagi/commcare-hq/pull/33597/commits/8781f14533f316d00fc9e04e704920d253b13423